tinygrad/helpers.py

```diff
@@ -76,6 +77,19 @@ def temp(x:str) -> str: return (pathlib.Path(tempfile.gettempdir()) / x).as_posi
 
 class GraphException(Exception): pass
 
+V = TypeVar("V")
+def bfs(graph:DefaultDict[T, List[T]], in_degree:DefaultDict[T, int], insert:Callable[[T, Dict[T, V]], None]) -> Dict[T, V]:
```

Comment:
isn't `in_degree` a function of `graph`, what if the two inputs disagree?

also feel this is a lot less flexible, like it's more difficult to experiment with a different traversing order

Reply:
I think it's worth the perf cost to not have to loop through the entire graph - Though I can make in_degree optional.
Will include cycle assert to make this safer.

Yeah the goal I started with was to library BFS specifically, rather than flexibility focusing on removing repetitiveness.